enhancements/cluster-api/installing-openshift-natively-via-cluster-api.md

```diff
@@ -0,0 +1,598 @@
+1. Integrate Hive directly with Cluster API to remove the need to execute the installer during Cluster Deployments.
+
+### Non-Goals
+
+1. Provisioning of the Cluster API control plane.[^2]
+1. Generation in-tree of ignition data for the bootstrap node.
+
+## Proposal
+
+A Cluster API control plane and bootstrap provider will be created to handle the orchestration and configuration of the OpenShift cluster during the bootstrap process.
+The control plane provider will be responsible for creating (and destroying) the bootstrap node, and provisioning the control plane nodes once the bootstrap node is ready.
+The bootstrap provider will be responsible for generating the correct ignition data for the bootstrap node, control plane nodes and worker nodes
+
+Where gaps exist in the Cluster API infrastructure providers, small, focused supplemental infrastructure providers will be created to fill those gaps.
+For example, on AWS, where Cluster API does not manage DNS, a supplemental provider will be created to manage DNS records for the cluster.
+
+Provide a configurable, opinionated and controller driven multi-cluster and single cluster provisioning experience:
+* Driven by install config and Cluster API resources
+* Cluster API resources generated by the installer for opinionated install
+* Self crafted Cluster API resources for custom install or install on external platforms
+* Customisation of bootstrap manifests via control plane provider configuration
+
+### Workflow Description
+
+#### Opinionated installer generated infrastructure definitions
+
+The installer binary will be updated to transform the existing install config into Cluster API resources.
+
+During the `manifests` phase, the installer will generate the following Cluster API resources into an `infrastructure-manifests` folder:
+* Cluster
+* OpenShiftControlPlane
+* InfrastructureCluster (e.g. `AWSCluster`)
+* InfrastructureMachineTemplate (e.g. `AWSMachineTemplate`)
+
+Additionally supplemental resources will be required:
+* Namespace (each cluster will be created in a unique namespace for isolation)
+* Infrastructure Credentials secret (to allow the Cluster API control plane to authenticate to the cloud provider).
+
+The installer will then diverge from the current path, which is to execute `ignition-configs` and then `cluster` phases.
+
+The `ignition-configs` phase will become a terminal phase and will be used by the bootstrap provider to generate the ignition configs for the bootstrap node, control plane nodes and worker nodes.
+No changes are expected to the `ignition-configs` phase and existing UPI workflows should continue to work.
+
+The `cluster` phase will now skip the `ignition-configs` phase and will instead apply the Cluster API resources generated in the `manifests` phase to the Cluster API control plane.
+
+The installer will directly apply the Cluster API resources to the Cluster API control plane.
```

Comment:
I think that's up to the installer team and oc folks to decide, but you're right, it is effectively just running an oc apply at this point to take manifests from the laptop or whatever and apply them to the control plane for CAPI.

I know there's prior art for the installer embedding binaries so it's possible we leverage and subprocess the oc binary for this purpose rather than re-inventing the wheel. This would align with a current exploration avenue of using subprocesses to run the temporary CAPI control plane for provisioning